aws_lambda_powertools/utilities/data_classes/code_pipeline_job_event.py

```diff
@@ -255,10 +274,53 @@ def get_artifact(self, artifact_name: str, filename: str) -> str | None:
         if artifact is None:
             return None
 
-        with tempfile.NamedTemporaryFile() as tmp_file:
-            s3 = self.setup_s3_client()
-            bucket = artifact.location.s3_location.bucket_name
-            key = artifact.location.s3_location.key
-            s3.download_file(bucket, key, tmp_file.name)
-            with zipfile.ZipFile(tmp_file.name, "r") as zip_file:
-                return zip_file.read(filename).decode("UTF-8")
+        s3 = self.setup_s3_client()
+        bucket = artifact.location.s3_location.bucket_name
+        key = artifact.location.s3_location.key
+
+        if filename:
+            with tempfile.NamedTemporaryFile() as tmp_file:
+                s3.download_file(bucket, key, tmp_file.name)
+                with zipfile.ZipFile(tmp_file.name, "r") as zip_file:
+                    return zip_file.read(filename).decode("UTF-8")
+
+        return s3.get_object(Bucket=bucket, Key=key)["Body"].read()
+
+    def put_artifact(self, artifact_name: str, body: Any, content_type: str) -> None:
+        """Writes an object to an s3 output artifact.
+
+        Parameters
+        ----------
+        artifact_name : str
+            Name of the S3 artifact to upload
+        body: Any
+            The data to be written. Binary files should use io.BytesIO.
+        content_type: str
+            The content type of the data.
+
+        Returns
+        -------
+        None
+        """
+        artifact = self.find_output_artifact(artifact_name)
+        if artifact is None:
+            raise ValueError(f"Artifact not found: {artifact_name}.")
+
+        s3 = self.setup_s3_client()
+        bucket = artifact.location.s3_location.bucket_name
+        key = artifact.location.s3_location.key
+        encryption_key_id = self.data.encryption_key.get_id
+        encryption_key_type = self.data.encryption_key.get_type
+
+        if encryption_key_type == "KMS":
+            encryption_key_type = "aws:kms"
```

Comment:
This is an artifact of different APIs/sources providing/requiring different formats of essentially the same value.